src/nvim/eval/funcs.c

```diff
@@ -3939,9 +3939,9 @@ dict_T *create_environment(const dictitem_T *job_env, const bool clear_env, cons
       size_t len = strlen(required_env_vars[i]);
       dictitem_T *dv = tv_dict_find(env, required_env_vars[i], (ptrdiff_t)len);
       if (!dv) {
-        const char *env_var = os_getenv(required_env_vars[i]);
-        if (env_var) {
-          tv_dict_add_str(env, required_env_vars[i], len, env_var);
+        if (os_env_exists(required_env_vars[i], true)) {
+          tv_dict_add_allocated_str(env, required_env_vars[i], len,
+                                    os_getenv(required_env_vars[i]));
```

Comment:
Wait, it seems that you can just change `tv_dict_add_str` to `tv_dict_add_allocated_str` in the original code.
```suggestion
        const char *env_var = os_getenv(required_env_vars[i]);
        if (env_var) {
          tv_dict_add_allocated_str(env, required_env_vars[i], len, env);
```

Reply:
`env_var` at the end, but yes :-)

Reply:
Done.